### PR TITLE
Adding ethograms and structures

### PR DESCRIPTION
**Pull request by rposborne**

### CODE REVIEW

File: app/models/ethogram.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  has_many :ethogram_structures
+
+  def current_ethogram_structure
+    ethogram_structures.order(:created_at).last
+  end
+
+  delegate :structure, to: :current_ethogram_structure, allow_nil: true
+
+  def structure_id
+    current_ethogram_structure.id
```
Comment (kwokster10) [inline app/models/ethogram.rb]: if we are allowing nil, `current_ethogram_structure&.id`

File: app/views/ethograms/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<p id="notice"><%= notice %></p>
```
Comment (kwokster10) [inline app/views/ethograms/index.html.erb]: we could move this into the application.html.erb file

File: spec/requests/ethograms_spec.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe "Ethograms", type: :request do
+  describe "GET /ethograms" do
+    it "works! (now write some real specs)" do
```
Comment (kwokster10) [inline spec/requests/ethograms_spec.rb:5]: That's great lol >_<

File: app/views/ethograms/index.html.erb (hunk 2 above)
Comment (rposborne) [reply app/views/ethograms/index.html.erb]: Awesome catch, I think this is a duplicate.

File: app/models/ethogram.rb (hunk 1 above)
Comment (rposborne) [reply app/models/ethogram.rb]: .... Yup, totes missed that case.

**Pull request merged**